### PR TITLE
utils/shebang: add rewrite_shebang (+ Perl & improved Python support)

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Example usage:

```rb
require "language/perl"

class A < Formula
  include Language::Perl

  # ...

  uses_from_macos "perl"

  def install
    # ...

    rewrite_perl_shebang

    # ...
  end
end
```

In the above example all matching shebangs will be replaced with `/usr/bin/perl` on macOS and `Formula["perl"].opt_bin/"perl"` on Linux. If `depends_on "perl"` is used, then `Formula["perl"].opt_bin/"perl"` will be used on both platforms.

### CODE REVIEW

File: Library/Homebrew/language/perl.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        raise "Cannot use rewrite_perl_shebang on a formula which does not depend on Perl."
+      end
+
+      regex = %r{^#! ?/usr/bin/(env )?perl$}
+      maximum_regex_length = 20 # the length of "#! /usr/bin/env perl"
+      Pathname.pwd.find do |f|
+        next unless f.file?
+        next unless regex.match?(f.read(maximum_regex_length))
+
+        Utils::Inreplace.inreplace f.to_s, regex, "#!#{perl_path}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/language/perl.rb]: This could share some code with the Python version.

**Comment by MikeMcQuaid** [comment]
> ```ruby
> rewrite_perl_shebang
> ```

I think I'd like to have some DSL (either by naming or ideally by arguments) that feels like less of a "black box". One of the main features of the Formula DSL is that someone reading through it should understand what's necessary to e.g. run instructions on their machine to build the software. This criticism could be applied to the Python version too, I appreciate, so it might not be a bad idea to adjust that accordingly too. CC @iMichka for thoughts.

**Comment by Bo98** [comment]
That makes sense. Equally, exposing even as much as `rewrite_python_shebang` does will mean conditional Mac & Linux checks, which would be ideal to avoid.

Do you have any suggestions of what it should look like from a formula file view? I could see `Pathname.pwd.find` stuff being lifted out of the DSL and into the formula file.

**Comment by MikeMcQuaid** [comment]
How about something like:

```ruby
buildpath.find { |f| rewrite_shebang(detected_perl_shebang, f) }
# or for a single file
rewrite_shebang(detected_perl_shebang, "somefile.pl")
```

and have `detected_perl_shebang` be a function that generates the perl shebang we want rewritten (and we could do the same thing for `detected_python_shebang` or something).

I'm not crazy on the names I'm suggesting above but just thinking how to make it as granular and self-explanatory as possible?

**Comment by Bo98** [comment]
Make sense to me.

**Comment by Bo98** [comment]
The regex & max length will need to incorporated somewhere for a generic `rewrite_shebang` - I'll have a think of the best way to do that.

**Comment by MikeMcQuaid** [comment]
`detected_perl_shebang` could return a `Hash` or even some object that would mean if you used it directly in the formula (without `rewrite_shebang) you might need to do `detected_perl_shebang.regex`, `detected_perl_shebang.max_length`, `detected_perl_shebang.string`, etc.?

**Comment by Bo98** [comment]
Yeah, a `Hash` seems sensible.

**Comment by Bo98** [comment]
I've made some tweaks just now.

**Comment by MikeMcQuaid** [comment]
@Bo98 Love it! Could I see an example formula usage (or, ideally, usage of both forms you suggested). After that would be good to add some basic tests and (if you can be bothered and I'll do it in another PR if you can't): refactor Python stuff to use this (while keeping existing naming).

**Comment by Bo98** [comment]
I tested this change on `pulledpork` (this is from a homebrew-core perspective - so no Linux checks were here):

```diff
diff --git a/Formula/pulledpork.rb b/Formula/pulledpork.rb
index f70634d07e..04953b9db0 100644
--- a/Formula/pulledpork.rb
+++ b/Formula/pulledpork.rb
@@ -1,4 +1,8 @@
+require "language/perl"
+
 class Pulledpork < Formula
+  include Language::Perl
+
   desc "Snort rule management"
   homepage "https://github.com/shirkdog/pulledpork"
   url "https://github.com/shirkdog/pulledpork/archive/v0.7.3.tar.gz"
@@ -33,7 +37,7 @@ class Pulledpork < Formula
       end
     end
 
-    inreplace "pulledpork.pl", "#!/usr/bin/env perl", "#!/usr/bin/perl"
+    rewrite_shebang(detected_perl_shebang, "pulledpork.pl")
 
     chmod 0755, "pulledpork.pl"
     bin.install "pulledpork.pl"
```

I've also just tried `percona-toolkit`:

```diff
diff --git a/Formula/percona-toolkit.rb b/Formula/percona-toolkit.rb
index 714e13d454..116c8bb6d9 100644
--- a/Formula/percona-toolkit.rb
+++ b/Formula/percona-toolkit.rb
@@ -1,4 +1,8 @@
+require "language/perl"
+
 class PerconaToolkit < Formula
+  include Language::Perl
+
   desc "Percona Toolkit for MySQL"
   homepage "https://www.percona.com/software/percona-toolkit/"
   url "https://www.percona.com/downloads/percona-toolkit/3.1.0/source/tarball/percona-toolkit-3.1.0.tar.gz"
@@ -53,12 +57,7 @@ class PerconaToolkit < Formula
     # Disable dynamic selection of perl which may cause segfault when an
     # incompatible perl is picked up.
     # https://github.com/Homebrew/homebrew-core/issues/4936
-    bin.find do |f|
-      next unless f.file?
-      next unless f.read("#!/usr/bin/env perl".length) == "#!/usr/bin/env perl"
-
-      inreplace f, "#!/usr/bin/env perl", "#!/usr/bin/perl"
-    end
+    bin.find { |f| rewrite_shebang(detected_perl_shebang, f) }
 
     bin.env_script_all_files(libexec/"bin", :PERL5LIB => ENV["PERL5LIB"])
   end
```

**Comment by Bo98** [comment]
I should probably try fix `brew tests` erroring on my machine before I touch it too much. I'll have a quick look at Python.

**Comment by MikeMcQuaid** [comment]
> I tested this change on `pulledpork` (this is from a homebrew-core perspective - so no Linux checks were here):

That looks perfect. Great work.
> I should probably try fix `brew tests` erroring on my machine before I touch it too much.

Shout if I can help with that. `brew tests --only` may be a useful workaround.

**Comment by Bo98** [comment]
I'll give `--only` a shot.

In terms of the rest, have you come across this error? If not, I'll look into it myself at some point.

<details>
<summary>Test output</summary>

```
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:661: [BUG] Segmentation fault at 0x0000000000000018
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin13]

-- Crash Report log information --------------------------------------------
   See Crash Report log file under the one of following:
     * ~/Library/Logs/DiagnosticReports
     * /Library/Logs/DiagnosticReports
   for more details.
Don't forget to include the above Crash Report log file in bug reports.

-- Control frame information -----------------------------------------------
c:0012 p:0008 s:0056 e:000053 BLOCK  /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:661 [FINISH]
c:0011 p:---- s:0049 e:000048 IFUNC 
c:0010 p:---- s:0046 e:000045 CFUNC  :each
c:0009 p:---- s:0043 e:000042 CFUNC  :each_with_index
c:0008 p:0006 s:0039 e:000038 METHOD /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660
c:0007 p:0024 s:0034 e:000033 METHOD /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1555
c:0006 p:0047 s:0028 e:000027 METHOD /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1421
c:0005 p:0006 s:0023 e:000022 BLOCK  /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:68 [FINISH]
c:0004 p:---- s:0020 e:000019 CFUNC  :synchronize
c:0003 p:0223 s:0016 e:000015 METHOD /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67
c:0002 p:0027 s:0006 e:000005 TOP    <internal:gem_prelude>:4 [FINISH]
c:0001 p:0000 s:0003 E:001520 (none) [FINISH]

-- Ruby level backtrace information ----------------------------------------
<internal:gem_prelude>:4:in `<internal:gem_prelude>'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb:68:in `block in gem'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1421:in `activate'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb:1555:in `add_self_to_load_path'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660:in `load_path_insert_index'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660:in `each_with_index'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:660:in `each'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb:661:in `block in load_path_insert_index'

-- Machine register context ------------------------------------------------
 rax: 0x000000000000008e rbx: 0x0000000109460d60 rcx: 0xfffffffffffffffe
 rdx: 0x0000000000000000 rdi: 0x00007ff23f826a00 rsi: 0x00007ff23f826250
 rbp: 0x00007ffee71f2410 rsp: 0x00007ffee71f2410  r8: 0x00007ff23dc00e28
  r9: 0x00007ff23f04dc68 r10: 0x0000000109460d60 r11: 0x00007ff23f827ac0
 r12: 0x00007ff23f827ac0 r13: 0x00007fffb5ffd070 r14: 0x00007ff23f826250
 r15: 0x00007ff23f826a00 rip: 0x0000000108c65054 rfl: 0x0000000000010246

-- C level backtrace information -------------------------------------------
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(rb_vm_bugreport+0x82) [0x108c6ac59]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108b33232) [0x108b33232]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108bfe9f4) [0x108bfe9f4]
/usr/lib/system/libsystem_platform.dylib(_sigtramp+0x1d) [0x7fff7fa70b5d]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c65054) [0x108c65054]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c4eb89) [0x108c4eb89]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c5cb8e) [0x108c5cb8e]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c587f0) [0x108c587f0]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108b2a34e) [0x108b2a34e]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c5fd32) [0x108c5fd32]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c617f5) [0x108c617f5]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c58558) [0x108c58558]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108adc518) [0x108adc518]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c565b0) [0x108c565b0]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c58c89) [0x108c58c89]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c58b36) [0x108c58b36]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c58b99) [0x108c58b99]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108b2729e) [0x108b2729e]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c66d8f) [0x108c66d8f]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c6625f) [0x108c6625f]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c4bb6b) [0x108c4bb6b]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c4eace) [0x108c4eace]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c5cb8e) [0x108c5cb8e]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c585c7) [0x108c585c7]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108b3b367) [0x108b3b367]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c66d8f) [0x108c66d8f]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c6625f) [0x108c6625f]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c4bb6b) [0x108c4bb6b]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c4eace) [0x108c4eace]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c5cb8e) [0x108c5cb8e]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108c72f94) [0x108c72f94]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108bfa2cc) [0x108bfa2cc]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108b3a2c0) [0x108b3a2c0]
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby(0x108a09747) [0x108a09747]

-- Other runtime information -----------------------------------------------

* Loaded script: /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby

* Loaded features:

    0 enumerator.so
    1 thread.rb
    2 rational.so
    3 complex.so
    4 encdb.so
    5 trans/transdb.so
    6 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/x86_64-darwin13/rbconfig.rb
    7 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/compatibility.rb
    8 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/defaults.rb
    9 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/deprecate.rb
   10 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/errors.rb
   11 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/version.rb
   12 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/requirement.rb
   13 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/platform.rb
   14 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/basic_specification.rb
   15 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/stub_specification.rb
   16 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/delegate.rb
   17 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/rfc2396_parser.rb
   18 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/rfc3986_parser.rb
   19 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/common.rb
   20 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/generic.rb
   21 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/file.rb
   22 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/ftp.rb
   23 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/http.rb
   24 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/https.rb
   25 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/ldap.rb
   26 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/ldaps.rb
   27 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri/mailto.rb
   28 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/uri.rb
   29 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification_policy.rb
   30 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/util/list.rb
   31 stringio.so
   32 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/specification.rb
   33 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/exceptions.rb
   34 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/util.rb
   35 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/bundler_version_finder.rb
   36 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/dependency.rb
   37 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_gem.rb
   38 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/monitor.rb
   39 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb
   40 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/core_ext/kernel_warn.rb
   41 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems.rb
   42 /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/lib/ruby/2.6.0/rubygems/path_support.rb

[NOTE]
You may have encountered a bug in the Ruby interpreter or extension libraries.
Bug reports are welcome.
For details: https://www.ruby-lang.org/bugreport.html

[IMPORTANT]
Don't forget to include the Crash Report log file under
DiagnosticReports directory in bug reports.

Error: testball: failed
```
</details>

**Comment by Bo98** [comment]
Made changes to Python so that both of these work:

```rb
Language::Python.rewrite_python_shebang(Formula["python@3.8"].opt_bin/"python3")
```
```rb
buildpath.find { |f| rewrite_shebang(Language::Python.detected_python_shebang(self), f) }
```

Unlike other language modules, `Language::Python` is not designed to be a mixin (`Language::Python::Virtualenv` is).

Maybe `Language::Perl` shouldn't either and the shebang stuff should be in a submodule of `Language::Perl`. `Language::Perl::Shebang`?

**Comment by MikeMcQuaid** [comment]
> In terms of the rest, have you come across this error?

Yeh, that's a crappy one where integration tests crash 😭. What's your `brew config` out of interest? I think I'll sort that out once and for all.

---

This looks really great to me, nice work. I might suggest that `Language::Python.rewrite_python_shebang` get moved under `compat/` so that we can deprecate it and replace its usage in formulae with your new version.

**Comment by Bo98** [comment]
```
HOMEBREW_VERSION: 2.2.11-8-g3955e7e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3955e7e13b980ad4f8d38ffc3c0253d86204902e
Last commit: 11 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 706b7f8533f5e5dba8b919019d9dcd5e6fd4a471
Core tap last commit: 4 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BINTRAY_KEY: set
HOMEBREW_BINTRAY_USER: bo98
HOMEBREW_DEV_CMD_RUN: 1
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby
Clang: 11.0 build 1100
Git: 2.21.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 11.0.6, 11.0.2, 1.8.0_242
macOS: 10.14.6-x86_64
CLT: 11.2.0.0.1.1571444319
Xcode: 11.3.1
XQuartz: 2.7.11 => /opt/X11
```

**Comment by Bo98** [comment]
You've probably noticed, but I've gone ahead and made `Language::Python::Shebang` (and similarly for Perl) so the Python example looks much better now:

```rb
buildpath.find { |f| rewrite_shebang(detected_python_shebang, f) }
```
(with `include Language::Python::Shebang` earlier)

**Pull request merged**